clippy_lints/src/if_not_else.rs

```diff
@@ -52,8 +52,12 @@ impl EarlyLintPass for IfNotElse {
         if in_external_macro(cx.sess(), item.span) {
             return;
         }
-        if let ExprKind::If(ref cond, _, Some(ref els)) = item.kind {
-            if let ExprKind::Block(..) = els.kind {
+        if let ExprKind::If(ref cond, ref thn, Some(ref els)) = item.kind {
+            // Don't suggest rewriting if the result would be top-heavy.
+            if let ExprKind::Block(ref els_block, ..) = els.kind {
+                if thn.stmts.len() < els_block.stmts.len() {
```

Comment:
Because we already talked about cost-benefit:

The cost for this configuration option would be way higher, since this:

```rust
if !a {
    println!("xyz");
} else {
    let _ = if {
        // ...100 lines
    };
}
```

would still lint. So we would need to implement a line counter. Or something that calculates complexity. But detecting complexity is a different story. Long story short: we can't do this (yet?, probably?, ...? https://github.com/rust-lang/rust-clippy/issues/3793#issuecomment-542457771)

So what @sanmai-NL said, the cost for a working implementation for this would be too high for the benefit that it would give for a pedantic lint IMO.